src/core/obj.js

```diff
@@ -252,6 +254,114 @@ var Catalog = (function CatalogClosure() {
+        var currentLabel = '', currentIndex = 1;
+
+        for (var i = 0, ii = this.numPages; i < ii; i++) {
+          if (nums.hasOwnProperty(i)) {
+            var labelDict = nums[i];
+            assert(isDict(labelDict), 'The PageLabel is not a dictionary.');
+
+            var type = labelDict.get('Type');
+            assert(!type || (isName(type) && type.name === 'PageLabel'),
+                   'Invalid type in PageLabel dictionary.');
+
+            var s = labelDict.get('S') || Name.get('');
+            assert(isName(s), 'Invalid style in PageLabel dictionary.');
+            style = s.name;
+
+            prefix = labelDict.get('P') || '';
+            assert(isString(prefix), 'Invalid prefix in PageLabel dictionary.');
+
+            start = labelDict.get('St') || 1;
+            assert(isInt(start), 'Invalid start in PageLabel dictionary.');
+            currentIndex = start;
+          }
+
+          switch (style) {
+            case 'D':
+              currentLabel = '' + currentIndex;
+              break;
+            case 'R':
+            case 'r':
+              var ROMAN_NUMBERS = [
+                '', 'C', 'CC', 'CCC', 'CD', 'D', 'DC', 'DCC', 'DCCC', 'CM',
+                '', 'X', 'XX', 'XXX', 'XL', 'L', 'LX', 'LXX', 'LXXX', 'XC',
+                '', 'I', 'II', 'III', 'IV', 'V', 'VI', 'VII', 'VIII', 'IX'
+              ];
+              var number = currentIndex, pos;
+              currentLabel = '';
+
+              while (number >= 1000) {
+                currentLabel += 'M';
+                number -= 1000;
+              }
+              // Hundreds
+              pos = (number / 100) | 0;
+              number %= 100;
+              currentLabel += ROMAN_NUMBERS[pos];
+              // Tens
+              pos = (number / 10) | 0;
+              number %= 10;
+              currentLabel += ROMAN_NUMBERS[10 + pos];
+              // Ones
+              currentLabel += ROMAN_NUMBERS[20 + number];
+              break;
+            case 'A':
+            case 'a':
+              var LIMIT = 26; // Use only the characters A--Z, or a--z.
+              var letterIndex = currentIndex - 1;
+
+              var character = String.fromCharCode((letterIndex % LIMIT) + 0x41);
+              var numChars = (letterIndex / LIMIT) | 0;
+              currentLabel = '';
+
+              for (var j = 0; j <= numChars; j++) {
+                currentLabel += character;
+              }
+              break;
+            default:
+              if (!style) {
+                currentLabel = '';
+                break;
+              }
+              error('Invalid style "' + style + '" in PageLabel dictionary.');
+          }
+
+          if (style === 'r' || style === 'a') {
+            currentLabel = currentLabel.toLowerCase();
+          }
+          currentIndex++;
+
+          pageLabels[i] = prefix + currentLabel;
+        }
+      }
+      return (pageLabels.length > 0 ? pageLabels : null);
```

Comment:
I was thinking that we could change this to `return pageLabels;` if we initialize `pageLabels` to `null` at https://github.com/Snuffleupagus/pdf.js/commit/e5d107ef97933ec0dd43ed800fc353fce3b3e8e6#diff-3fd0fc66c9db990d996e595419b3f1e1R272. We can then make it an array in the if-statement at https://github.com/Snuffleupagus/pdf.js/commit/e5d107ef97933ec0dd43ed800fc353fce3b3e8e6#diff-3fd0fc66c9db990d996e595419b3f1e1R275, which would also save an array allocation for PDFs that do not have page labels, which will likely be most of the existing PDFs.